mlflow/bedrock/stream.py

```diff
@@ -65,12 +66,90 @@ def _end_span(self):
 class InvokeModelStreamWrapper(BaseEventStreamWrapper):
     """A wrapper class for a event stream returned by the InvokeModelWithResponseStream API."""
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._usage_buffer = {}
+
     @capture_exception("Failed to handle event for the stream")
     def _handle_event(self, span, event):
         chunk = json.loads(event["chunk"]["bytes"])
         self._span.add_event(SpanEvent(name=chunk["type"], attributes={"json": json.dumps(chunk)}))
 
+        # Buffer usage information from streaming chunks
+        self._buffer_usage(chunk)
+
+    def _buffer_usage(self, chunk: dict):
+        """Buffer token usage information from streaming chunks."""
+
+        def _extract_and_buffer_tokens(usage: dict):
+            # For streaming, we need to handle partial updates where only some token counts
+            # are provided. Extract individual token counts and update buffer directly.
+
+            def _pick(d: dict, *names):
+                for n in names:
+                    if n in d:
+                        return d[n]
+                return None
```

Comment:
```suggestion
            def _pick(d: dict[str, Any], names: list[str]) -> int:
                return next((d[name] for name in names if name in d), None)
```

Let's not use variable-length argument if it's not necessary.